provision/acc_provision/templates/aci-containers.yaml

```diff
@@ -2442,7 +2442,7 @@ spec:
       {% if config.kube_config.use_system_node_priority_class %}
       priorityClassName: system-node-critical
       {% else %}
-      {% if not config.kube_config.no_priority_class %}
+      {% if config.kube_config.no_priority_class %}
```

Comment:
why this change - here it is saying that the no_priority_class is true we use system-cluster-critical but ideally when it is false then we use the priorityclass. Is this a valid check ?

Reply:
Purpose of this flag is to set priority class for k8s platform.
1) Either we can continue to use the same name (to go with minimal changes) or
2) Use different flag name and replace 'no_priority_class' flag with the new name at all the places in
    k8s etc flavors / wherever required

@abhis2112 @snaiksat

Reply:
@abhis2112 Only use_system_node_priority_class is expected from user, other two no_priority_class and use_acicni_priority_class are coming from flavors.yaml and its always going to be present.

When a new flavor is added it's expected and mandatory to add no_priority_class(for k8s) / use_acicni_priority_class (for ocp). If it's not added then it's an error and must be resolved during the flavor addition itself. Other than this I don't see any issue.

If I am correct, you are suggesting we add a default priority class if no option provided at all (we have to decide which one is common across all platforms), but do we really need that with above mentioned approach ?

Reply:
so if I deduce it right , in case of k8s flavor if the mandatory field is added (no_priority_class) then according to old logic the if check will always be false since no_priority_class will always be present as per flavor then we will end up with no priority class name configuration.

With the new change, the logic makes it default - with no_priority_class (always set) use system_cluster_critical priorityclassname  . So the question to you is - is there a default priorityclass earlier for k8s or is it expected with this change

Reply:
There is no default priority class earlier also.

Logic used to set priority class for k8s and ocp flavours is little confusing.

1) for k8s - not adding flag `no_priority_class` is the way 

   flavors.yaml ->

<img width="454" alt="Screenshot 2024-02-29 at 8 40 59 PM" src="https://github.com/noironetworks/acc-provision/assets/8308775/bcfc7fa9-22cc-4a76-b70b-9bc322c68765">

aci-containers.yaml (https://github.com/noironetworks/acc-provision/blob/master/provision/acc_provision/templates/aci-containers.yaml#L2474)

```
{% if not config.kube_config.no_priority_class %}
      priorityClassName: system-cluster-critical
{% endif %}

```
2) for ocp - setting both `no_priority_class` and `use_acicni_priority_class` is the way

    flavors.yaml ->

<img width="576" alt="Screenshot 2024-02-29 at 8 42 42 PM" src="https://github.com/noironetworks/acc-provision/assets/8308775/229a8b4f-f2d5-4d53-a58b-ea915b18d97d">

aci-containers.yaml (https://github.com/noironetworks/acc-provision/blob/master/provision/acc_provision/templates/aci-containers.yaml#L2477)

```
      {% if config.kube_config.use_acicni_priority_class %}
      priorityClassName: acicni-priority
      {% endif %}
```

Setting both the flags to determine ocp flavor and then not setting any flag to determine k8s flavors is confusing. So straight forward logic should be you set the flag itself for corresponding flavor.